reference/conanfile/tools/microsoft.rst

```diff
@@ -625,21 +625,118 @@ This generator can be used as:
 
     class Pkg(ConanFile):
         settings = "os", "compiler", "build_type", "arch"
-
-        # attribute declaration
         generators = "NMakeToolchain"
 
-        # OR explicit usage in the generate() method
+        def build(self):
+            self.run("nmake /f makefile")
+
+Or it can be fully instantiated in the conanfile ``generate()`` method:
+
+.. code:: python
+
+    from conan import ConanFile
+    from conan.tools.microsoft import NMakeToolchain
+
+    class Pkg(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+
         def generate(self):
-            toolchain = NMakeToolchain(self)
-            toolchain.generate()
+            tc = NMakeToolchain(self)
+            tc.generate()
 
         def build(self):
-            self.run(f"nmake /f makefile")
+            self.run("nmake /f makefile")
+
+NMakeToolchain generator will create a ``conannmaketoolchain.bat`` environment script injecting flags
+deduced from profile (build_type, runtime, cppstd, build flags from conf) into environment variables
+NMake can understand: ``CL``, ``_LINK_``, ``CFLAGS``, ``CPPFLAGS`` and ``CXXFLAGS``.
```

Comment:
actually `CFLAGS`, `CPPFLAGS` and `CXXFLAGS` have been removed from https://github.com/conan-io/conan/pull/12665

Reply:
Oops, that is true. If you can, please do a new PR to ``develop``, otherwise, I'll try to revise it later today

Reply:
https://github.com/conan-io/docs/pull/2930